inference-engine/tests/functional/inference_engine/ngraph_reader/non_max_suppression_tests.cpp

```diff
@@ -29,55 +29,62 @@ TEST_F(NGraphReaderTests, ReadNonMaxSuppression) {
                 </port>
             </output>
         </layer>
-        <layer id="2" name="max_output_boxes_per_class" precision="I64" type="Const">
+        <layer id="2" name="max_output_boxes_per_class" precision="I64" type="Const" version="opset1">
             <data offset="0" size="8"/>
             <output>
-                <port id="0"/>
+                <port id="0" precision="I64"/>
             </output>
         </layer>
-        <layer id="3" name="iou_threshold" precision="FP32" type="Const">
+        <layer id="3" name="iou_threshold" precision="FP32" type="Const" version="opset1">
             <data offset="8" size="4"/>
             <output>
-                <port id="0"/>
+                <port id="0" precision="FP32"/>
             </output>
         </layer>
-        <layer id="4" name="score_threshold" precision="FP32" type="Const">
+        <layer id="4" name="score_threshold" precision="FP32" type="Const" version="opset1">
             <data offset="12" size="4"/>
             <output>
-                <port id="0"/>
+                <port id="0" precision="FP32"/>
+            </output>
+        </layer>
+        <layer id="5" name="soft_nms_sigma" precision="FP32" type="Const" version="opset1">
+            <data offset="16" size="4"/>
+            <output>
+                <port id="0" precision="FP32"/>
             </output>
         </layer>
-        <layer id="5" name="nms" type="NonMaxSuppression">
+        <layer id="6" name="nms" type="NonMaxSuppression" version="opset5">
```

Comment:
Please, add a test for NMS-1 and NMS-3 operations. We need to make sure that IR reader correctly works for all versions of the NMS.

Reply:
We cannot  add such tests for NMS-1, NMS-3, or NMS-4. The reason is the following. When IRv10 model contains NMS-1, NMS-3, or NMS-4, we read the network, and then `CNNNetworkImpl` constructor performs all nGraph transformations, and converts a resulting graph to legacy IR. In particular, NMS-1, NMS-3, NMS-4 are converted into NMS-5, and then NMS-5 will be converted into the inner operation NMSIE3. According to specifications, each of operations NMS-1, NMS-3, NMS-4 has one output port, but NMS-5 has three output ports, and one or two output ports can be disconnected. Hence, we will get legacy IR with `CNNLayer` having  `type = "NonMaxSuppression"` with three output ports and with one element in `OutputDataMap`. Hence, we can create neither reference legacy IR with one output, nor legacy IR with three outputs, because the function that compares instances of `CNNLayer`, uses a comparison of the number of all output ports, not only connected ports. Moreover, we cannot creates CNNNetworks from IRs V10 with NMS-5, because output shapes of NMS-5 are dynamic.